src/librustc_typeck/check/cast.rs

```diff
@@ -438,13 +504,16 @@ impl<'a, 'tcx> CastCheck<'tcx> {
                     Ok(s) => {
                         err.span_suggestion(
                             self.cast_span,
-                            "try casting to a `Box` instead",
+                            "you can cast to a `Box` instead",
                             format!("Box<{}>", s),
                             Applicability::MachineApplicable,
                         );
                     }
                     Err(_) => {
-                        err.span_help(self.cast_span, &format!("did you mean `Box<{}>`?", tstr));
+                        err.span_help(
+                            self.cast_span,
+                            &format!("you might have meatn`Box<{}>`", tstr),
```

Comment:
```suggestion
                            &format!("you might have meant `Box<{}>`", tstr),
```

I didn't see any tests which hit this branch?

Reply:
You're right, there was no test for it because it is the `span_to_snippet` recovery codepath (which is hard to test for).